dashboard/dashboard/pages/patients/__init__.py

```diff
@@ -238,8 +300,160 @@ def set_age_breakdown_buttons(
             pos_neg_outline,
         )
 
+    @app.callback(
+        Output("timeseries-plot", "children"),
+        Output("button_timeseries_all_data", "outline"),
+        Output("button_timeseries_train_val", "outline"),
+        Output("button_timeseries_pos_neg", "outline"),
+        Input("button_timeseries_all_data", "n_clicks"),
+        Input("button_timeseries_train_val", "n_clicks"),
+        Input("button_timeseries_pos_neg", "n_clicks"),
+    )
+    def set_timeseries_buttons(
+        n_clicks_all_data, n_clicks_train_val, n_clicks_pos_neg
+    ):
+        changed_id = [p["prop_id"] for p in dash.callback_context.triggered][0]
+        all_data_outline = train_val_outline = pos_neg_outline = True
+        if changed_id == "button_timeseries_all_data.n_clicks":
+            group = "all"
+            all_data_outline = False
+        elif changed_id == "button_timeseries_train_val.n_clicks":
+            group = "train_val"
+            train_val_outline = False
+        elif changed_id == "button_timeseries_pos_neg.n_clicks":
+            group = "pos_neg"
+            pos_neg_outline = False
+        else:
+            group = "all"
+            all_data_outline = False
+        
+        return (
+            create_patient_timeseries(data, group),
+            all_data_outline,
+            train_val_outline,
+            pos_neg_outline,
+        )
+
     return app
 
+def create_patient_timeseries(data, group):
+    def aggregate_timeseries(df):
+        timeseries = (
+            df.groupby("all_swab_dates").count()["Pseudonym"]
+            .groupby(pd.Grouper(freq="W"))
+            .sum()
+            .fillna(0)
+            .sort_index()
+        )
+        # Extend time series on front and back
+        extra = pd.Series(
+            [0, 0], index=["2020-02-01", pd.to_datetime("today")]
```

Comment:
This might need to be synchronised between pages, as I'm not sure how confusing it is, if some charts start at one time, and others another. I think the date here is justified, though, only bringing up the discussion.

Reply:
hmm... yeah I see your point but I think the different start dates make sense. One refers to when the pandemic started and patients enter hospital and the other refers to with the NCCID started data collection. Maybe extending the other one back with a little note to let users know that data collection started in May is the best option.